### PR TITLE
Fix crash when snarfing empty selection using mouse chord

### LINKED ISSUE #230: Crash when snarfing empty selection using mouse chord
To reproduce, cut and paste using mouse chord, without moving the mouse cursor. It doesn't crash if the cursor is at the beginning of the file.
```
panic: acme: textsetselect p0=315 p1=0 q0=315 q1=0 t.org=0 nchars=906

goroutine 38 [running, locked to thread]:
main.(*Text).SetSelect(0xc000194518, 0x13b, 0x0)
	/home/fhs/go/src/github.com/rjkroege/edwood/text.go:1288 +0x396
main.(*Text).Select(0xc000194518)
	/home/fhs/go/src/github.com/rjkroege/edwood/text.go:1176 +0x49e
main.MovedMouse(0x2ce, 0x10a, 0x1, 0x3f20b8ea)
	/home/fhs/go/src/github.com/rjkroege/edwood/acme.go:422 +0x48b
main.mousethread(0x69dde0, 0xc00009c118)
	/home/fhs/go/src/github.com/rjkroege/edwood/acme.go:301 +0x304
created by main.main.func1
	/home/fhs/go/src/github.com/rjkroege/edwood/acme.go:173 +0x82b
```

### PR DESCRIPTION
**Pull request by fhs**
Fixes #230

### CODE REVIEW

File: file.go [hunk 1]
```diff
@@ -533,7 +533,7 @@ func (f *File) Seq() int {
 // This does not align with the semantics of undo.Buffer.
 // Each "Mark" needs to have a seq value provided.
 // TODO(rjk): Consider providing the target seq value as an argument.
-func (f *File) Undo(isundo bool) (q0p, q1p int) {
+func (f *File) Undo(isundo bool) (q0p, q1p int, ok bool) {
```
Comment (rjkroege) [inline file.go]: It would be good here to have some kind of explanation of what `ok` means via comment?

**Comment by rjkroege** [comment]
It's not obvious why this patch is the right fix for #230 so it would be great to add some additional explanation to the commit? I agree that `Undo` should probably indicate that the returned `Buffer` offsets are meaningful.

I am unclear why `Text.Select` calls `Undo` at all (text.go:1175) other than short circuiting the chord cut/paste sequence in the same buffer.

**Comment by rjkroege** [comment]
Also: nice detective work on this issue.

**Comment by fhs** [comment]
I've updated the commit message. Basically, this is how it was done in the original C code.

File: file.go (hunk 1 above)
Comment (fhs) [reply file.go]: Done

**Pull request merged**